capacity4more/modules/c4m/features_og/c4m_features_og_manage/c4m_manage_categories.inc

```diff
@@ -0,0 +1,834 @@
+  drupal_set_message(t('Deleted term %name.', array('%name' => $form_state['values']['name'])));
+  watchdog('taxonomy', 'Deleted term %name.', array('%name' => $form_state['values']['name']), WATCHDOG_NOTICE);
+  $menu_item = menu_get_item(current_path());
+  $path = $menu_item['path'] == 'manage/categories/types' ? 'types' : '';
+  $form_state['redirect'] = 'manage/categories/' . $path;
+  cache_clear_all();
+  return;
+}
+
+/**
+ * Form builder to confirm resetting a terms under a specific category to alphabetical order.
+ *
+ * @ingroup forms
+ * @see taxonomy_vocabulary_confirm_reset_alphabetical()
+ */
+function c4m_taxonomy_vocabulary_confirm_reset_alphabetical($form, &$form_state, $vid, $tid) {
+  $vocabulary = taxonomy_vocabulary_load($vid);
+  $form['type'] = array('#type' => 'value', '#value' => 'vocabulary');
+  $form['vid'] = array('#type' => 'value', '#value' => $vid);
+  $form['machine_name'] = array('#type' => 'value', '#value' => $vocabulary->machine_name);
+  $form['reset_alphabetical'] = array('#type' => 'value', '#value' => TRUE);
+  $path = $tid ? $vocabulary->machine_name : 'types';
+
+  if ($tid) {
+    $term = taxonomy_term_load($tid);
+    $form['parent_tid'] = array('#type' => 'value', '#value' => $tid);
+    $form['name'] = array('#type' => 'value', '#value' => $term->name);
+
+    return confirm_form($form,
+      t('Are you sure you want to reset the children of the term %title to alphabetical order?',
+        array('%title' => $term->name)),
+      'manage/categories/' . $path,
+      '<div class="question">'
+      . t('Are you sure you want to reset the children of the term %title to alphabetical order?',
+        array('%title' => $term->name))
+      . '</div>'
+      . t('Resetting the children of the term %title will discard all custom ordering and sort items alphabetically.',
+        array('%title' => $term->name)),
+      t('Reset to alphabetical'),
+      t('Cancel'));
+  }
+  else {
+    return confirm_form($form,
+      t('Are you sure you want to reset the Categories to alphabetical order?'),
+      'manage/categories/' . $path,
+      '<div class="question">'
+      . t('Are you sure you want to reset the Categories to alphabetical order?')
+      . '</div>'
+      . t('Resetting the the Categories will discard all custom ordering and sort items alphabetically.'),
+      t('Reset to alphabetical'),
+      t('Cancel'));
+  }
+}
+
+/**
+ * Submit handler to reset a specific category's children to alphabetical order after confirmation.
+ *
+ * @see c4m_taxonomy_vocabulary_confirm_reset_alphabetical()
+ */
+function c4m_taxonomy_vocabulary_confirm_reset_alphabetical_submit($form, &$form_state) {
+  $params = array(
+    ':weight' => 0,
+    ':vid' => $form_state['values']['vid'],
+  );
+  $condition = '';
+  if (isset($form_state['values']['parent_tid'])) {
+    $params[':parent_tid'] = $form_state['values']['parent_tid'];
+    $condition = ' AND h.parent = :parent_tid';
+  }
+
+  db_query(
```

Comment:
Use the [db_update](https://api.drupal.org/api/drupal/includes!database!database.inc/function/db_update/7) function to get an [UpdateQuery](https://api.drupal.org/api/drupal/includes%21database%21query.inc/class/UpdateQuery/7) object. Use that object to dynamically create the update query and to execute it.